opening_hours/tests/test_hauki_resource_hash_updater.py

```diff
@@ -0,0 +1,203 @@
+import datetime
+
+import freezegun
+import pytest
+
+from opening_hours.models import OriginHaukiResource, ReservableTimeSpan
+from opening_hours.utils.hauki_api_client import HaukiAPIClient
+from opening_hours.utils.hauki_resource_hash_updater import HaukiResourceHashUpdater
+from opening_hours.utils.reservable_time_span_client import NEVER_ANY_OPENING_HOURS_HASH, ReservableTimeSpanClient
+from tests.decorators import patch_method
+from tests.factories.opening_hours import OriginHaukiResourceFactory, ReservableTimeSpanFactory
+
+# Applied to all tests
+pytestmark = [
+    pytest.mark.django_db,
+]
+
+############
+# __init__ #
+############
+
+
+def test__HaukiResourceHashUpdater__init__no_params(reservation_unit):
+    updater = HaukiResourceHashUpdater()
+
+    assert list(updater.hauki_resource_ids) == [
+        reservation_unit.origin_hauki_resource_id,
+        reservation_unit.unit.origin_hauki_resource_id,
+    ]
+
+
+def test__HaukiResourceHashUpdater__init__pass_resource_ids(reservation_unit):
+    updater = HaukiResourceHashUpdater([1, 2, 3])
+
+    assert updater.hauki_resource_ids == [1, 2, 3]
+
+
+def test__HaukiResourceHashUpdater__init__pass_empty_list(reservation_unit):
+    updater = HaukiResourceHashUpdater([])
+
+    assert updater.hauki_resource_ids == []
+
+
+##########################
+# _fetch_hauki_resources #
+##########################
+
+
+@patch_method(HaukiAPIClient.get_resources, return_value={"results": ["foo"]})
+def test__HaukiResourceHashUpdater__fetch_hauki_resources__single_page(reservation_unit):
+    updater = HaukiResourceHashUpdater()
+    updater._fetch_hauki_resources()
+
+    assert HaukiAPIClient.get_resources.call_count == 1
+    assert updater.fetched_hauki_resources == ["foo"]
+
+
+# First request uses `get_resources`
+@patch_method(HaukiAPIClient.get_resources, return_value={"results": ["foo"], "next": "page2"})
+# Later requests use `get`
+
```

Comment:
Extra space here is a bit strange, seems like the decorator above is just "floating" even though it's attached to the test function.